### PR TITLE
feat(alpha): monitoring task add source and sourceId props - [AH-1863]

### PR DESCRIPTION
**Pull request by nxame**
Updating type of monitoring task to accept `source` and `sourceId` optional props.

Bonus:
* Fix broken links in docs

### CODE REVIEW

**Review by polomani — APPROVED** [review]
LGTM

**Review by andersfylling — APPROVED** [review]

**Comment by polomani** [comment]
@absqueued you may try `<!-- markdown-link-check-disable -->` because 
`curl curl https://developer.cognite.com/dev/guides/iam/external-application.html#tokens` returns 404 even though the page exists.

the root problem is in docs portal

File: guides/auth-wrapper.md [hunk 1]
```diff
@@ -441,4 +441,4 @@ await client.authenticate(); // Returns the received token
 ​
 ## Next steps
 ​
-To read more about authentication, see [authentication process](https://doc.cognitedata.com/dev/guides/iam/external-application.html#tokens).
+To read more about authentication, see [authentication process](https://developer.cognite.com/dev/guides/iam/external-application.html#tokens).
```
Comment (polomani) [inline guides/auth-wrapper.md]:
```suggestion
To read more about authentication, see [authentication process](https://developer.cognite.com/dev/guides/iam/external-application#tokens).
```

File: guides/authentication.md [hunk 2]
```diff
@@ -181,4 +181,4 @@ const client = new CogniteClient({
 
 ## More
 
-Read more about the [authentication process](https://doc.cognitedata.com/dev/guides/iam/external-application.html#tokens).
+Read more about the [authentication process](https://developer.cognite.com/dev/guides/iam/external-application.html#tokens).
```
Comment (polomani) [inline guides/authentication.md]:
```suggestion
Read more about the [authentication process](https://developer.cognite.com/dev/guides/iam/external-application#tokens).
```

File: packages/stable/CHANGELOG.md [hunk 3]
```diff
@@ -695,7 +695,7 @@ Co-authored-by: Vegard Økland <vegard.okland@cognite.com>
   - TimeseriesClass
   - TimeSeriesList
 
-For replacements, see [MIGRATION_GUIDE_2xx_3xx.md](https://docs.cognite.com/dev/guides/sdk/js/migration.html#upgrade-javascript-sdk-2-x-to-3-x)
+For replacements, see [MIGRATION_GUIDE_2xx_3xx.md](https://docs.cognite.com/dev/guides/sdk/js/migration.html#upgrade-javascript-sdk-2x-to-3x)
```
Comment (polomani) [inline packages/stable/CHANGELOG.md]:
```suggestion
For replacements, see [MIGRATION_GUIDE_2xx_3xx.md](https://docs.cognite.com/dev/guides/sdk/js/migration#upgrade-javascript-sdk-2x-to-3x)
```

File: packages/stable/CHANGELOG.md [hunk 4]
```diff
@@ -695,7 +695,7 @@ Co-authored-by: Vegard Økland <vegard.okland@cognite.com>
   - TimeseriesClass
   - TimeSeriesList
 
-For replacements, see [MIGRATION_GUIDE_2xx_3xx.md](https://docs.cognite.com/dev/guides/sdk/js/migration.html#upgrade-javascript-sdk-2-x-to-3-x)
+For replacements, see [MIGRATION_GUIDE_2xx_3xx.md](https://developer.cognite.com/sdks/js/migration.html#upgrade-javascript-sdk-2x-to-3x)
```
Comment (polomani) [inline packages/stable/CHANGELOG.md]:
```suggestion
For replacements, see [MIGRATION_GUIDE_2xx_3xx.md](https://developer.cognite.com/sdks/js/migration/#upgrade-javascript-sdk-2x-to-3x)
```

**Review by saagar-cognite — APPROVED** [review]

**Pull request merged**